Tools/Pipeline/Common/PipelineController.FileWatcher.cs

```diff
@@ -0,0 +1,67 @@
+﻿// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using System.IO;
+using System.Threading;
+
+namespace MonoGame.Tools.Pipeline
+{
+    internal partial class PipelineController
+    {
+        private class FileWatcher
+        {
+            Thread _thread;
+
+            PipelineController _controller;
+            IView _view;
+
+            public FileWatcher(PipelineController controller, IView view)
+            {
+                _controller = controller;
+                _view = view;
+            }
+
+            public void Run()
+            {
+                Stop();
+
+                _thread = new Thread(new ThreadStart(ExistsThread));
+                _thread.Start();
+            }
+
+            public void Stop()
+            {
+                if (_thread != null)
+                    _thread.Abort();
```

Comment:
Probably should null out the _thread here too. otherwise we might call _thread.Abort() twice